component/read/read.go

```diff
@@ -39,24 +39,26 @@ const (
 	logTimestampProp   = "timestamp"
 	logEventSchemaProp = "schema"
 	logEventTableProp  = "table"
+
+	msgProtocol = "json"
 )
 
 type newBean func(fields []string, rows [][]interface{}) interface{}
 
 type Read struct {
-	dbMetadata *database.DbMetadata
+	dbMetadata database.DbMetadata
 	batcher    b.MqBatcher[b.MessageArguments]
-	conf       Connection
+	conf       conf.Connection
 	alias      string
 }
 
-func NewRead(batcher b.MqBatcher[b.MessageArguments], conf Connection, schemas map[string][]string, alias string) (*Read, error) {
-	dbMetadata, err := database.CreateMetadata(conf.Host, conf.Port, conf.Username, conf.Password, schemas)
+func NewRead(batcher b.MqBatcher[b.MessageArguments], conf conf.Connection, schemas conf.SchemasConf, alias string) (*Read, error) {
+	dbMetadata, err := database.LoadMetadata(conf.Host, conf.Port, conf.Username, conf.Password, schemas)
 	if err != nil {
-		return nil, errors.New("connect to database failure")
+		return nil, fmt.Errorf("loading schema metadata ta failure: %w", err)
 	}
 	return &Read{
-		dbMetadata: dbMetadata,
+		dbMetadata: *dbMetadata,
```

Comment:
note: If we need to dereference the pointer we are probably doing something wrong. Most likely, we can change `database.LoadMetadata` to return value instead of a pointer